vault/resource_identity_group.go

```diff
@@ -291,16 +291,18 @@ func identityGroupExists(d *schema.ResourceData, meta interface{}) (bool, error)
 	}
 
 	id := d.Id()
+
+	path := identityGroupIDPath(id)
 	key := id
 
+	// use the name if no ID is set
 	if len(id) == 0 {
-		return false, nil
-	} else {
 		key = d.Get("name").(string)
+		path = identityGroupNamePath(key)
 	}
 
 	log.Printf("[DEBUG] Checking if IdentityGroup %q exists", key)
-	resp, err := readIdentityGroup(client, id, true)
+	resp, err := client.Logical().Read(path)
```

Comment:
This change will break the client controlled consistency (CCC) bits. We may need to take a closer look at the issue.

Reply:
Since we actually do no longer are supporting `schema.Resource.Exists`, we can just call `d.SetId("")` in `identityGroupRead()` when `resp` is nil.

Reply:
Done! Thanks @benashz